subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/GenerateMavenPom.java

```diff
@@ -42,6 +43,7 @@
  *
  * @since 1.4
  */
+@DisableCachingByDefault(because = "Gradle doesn't understand the data structures")
```

Comment:
I think we should clarify what data structures are meant here. Something like:

```suggestion
@DisableCachingByDefault(because = "Gradle doesn't understand the data structures used to configure this task")
```